packages/ubuntu_bootstrap/lib/services/installer_service.dart

```diff
@@ -33,16 +36,28 @@ class InstallerService {
 
   Future<void> load() async {
     await monitorStatus().firstWhere((s) => s?.isLoading == false);
-    _pages ??=
-        await _client.getInteractiveSections() ?? await _config?.getPages();
+    _subiquityPages = (await _client.getInteractiveSections())?.toSet();
+    _excludedPages = _pageConfig?.excludedPages;
+
+    if ((_subiquityPages?.isNotEmpty ?? false) &&
+        (_excludedPages?.isNotEmpty ?? false)) {
+      final requiredExcludedPages =
+          _excludedPages!.where(_subiquityPages!.contains);
+      _log.info(
+          '$requiredExcludedPages are required by subiquity and cannot be excluded!');
+      _excludedPages!.removeWhere(requiredExcludedPages.contains);
```

Comment:
And then you can do removeAll with the intersection set here

Reply:
Thanks, good point! Didn't think about what further improvements I could make when I changed those types from lists to sets